src/Viewer.tsx

```diff
@@ -107,9 +108,15 @@ function Viewer(): ReactElement {
     return canvas;
   });
 
-  const [collection, setCollection] = useState<Collection | undefined>();
-  const [dataset, setDataset] = useState<Dataset | null>(null);
-  const [datasetKey, setDatasetKey] = useState("");
+  const collection = useViewerStateStore((state) => state.collection);
+  const setCollection = useViewerStateStore((state) => state.setCollection);
+  const dataset = useViewerStateStore((state) => state.dataset);
+  const datasetKey = useViewerStateStore((state) => state.datasetKey);
+  const setDataset = useViewerStateStore((state) => state.setDataset);
+  /** Backdrop key is null if the dataset has no backdrops, or during initialization. */
+  const selectedBackdropKey = useViewerStateStore((state) => state.backdropKey);
+  const setSelectedBackdropKey = useViewerStateStore((state) => state.setBackdropKey);
+
```

Comment:
It's also possible to do something like this:
```
const state = useViewerStateStore(useShallow((state) => ({
      collection: state.collection,
      dataset: dataset.collection,
      datasetKey: dataset.datasetKey,
    })
  )
);
```

If anything though, this is *more* lines than line-by-line selecting the relevant state.